### PR TITLE
NO-ISSUE: *: implement resource monitor api

### PR DESCRIPTION
**Pull request by hexfusion**
This PR implemented #329.

- add memory monitor
- improved CPU and Disk monitoring
- add fleet and standalone device support for defining resource alerts
- add resource controller
- implement support for alert rules

e2e will be a followup

### CODE REVIEW

File: api/v1alpha1/openapi.yaml [hunk 1]
```diff
@@ -17,7 +17,8 @@ paths:
           description: Redirect
           headers:
             Location:
-              type: "string"
+              schema:
+                type: string
```
Comment (hexfusion) [inline api/v1alpha1/openapi.yaml:21]: this was failing validation from previous change.

File: internal/agent/device/resource/resource.go [hunk 2]
```diff
@@ -62,18 +55,90 @@ func (m *ResourceManager) Run(ctx context.Context) {
...
-	CPUUsage  *CPUUsage
+type AlertSeverity struct {
+	Status v1alpha1.DeviceResourceStatusType
+	Level  int
+}
+
+// GetHighestSeverityResourceStatusFromAlerts returns the highest severity statusDeviceResourceStatusType from a list of alerts.
+func GetHighestSeverityResourceStatusFromAlerts(alerts []v1alpha1.ResourceAlertRule) (v1alpha1.DeviceResourceStatusType, error) {
+	if len(alerts) == 0 {
+		return v1alpha1.DeviceResourceStatusUnknown, nil
```
Comment (hexfusion) [inline internal/agent/device/resource/resource.go:129]: @celdrake

**Review by fzdarsky — APPROVED** [review]

File: internal/agent/device/resource/resource.go (hunk 2 above)
Comment (celdrake) [reply internal/agent/device/resource/resource.go:129]: Looks good, in the UI we did something similar to extract the message of the alert with the highest severity.
https://github.com/flightctl/flightctl-ui/pull/22/files#diff-e99c39011bc712d2766b296021bd3a9fac077341b017cbd2568455b6c7b4df47R40

**Review by mangelajo — APPROVED** [review]
Looks good, we could just follow up with tests while we merge it IMHO.

**Comment by hexfusion** [comment]
I want to hold for #347

**Pull request merged**